validators/cgroup_validator.go

```diff
@@ -38,8 +41,26 @@ func (c *CgroupsValidator) Name() string {
 
 const (
 	cgroupsConfigPrefix = "CGROUPS_"
+	unifiedMountpoint   = "/sys/fs/cgroup"
 )
 
+var (
+	isUnifiedOnce sync.Once
+	isUnified     bool
+)
+
+// IsCgroup2UnifiedMode returns whether we are running in cgroup v2 unified mode.
+func IsCgroup2UnifiedMode() bool {
+	isUnifiedOnce.Do(func() {
+		var st unix.Statfs_t
+		if err := unix.Statfs(unifiedMountpoint, &st); err != nil {
+			panic("cannot statfs cgroup root")
+		}
+		isUnified = st.Type == unix.CGROUP2_SUPER_MAGIC
+	})
+	return isUnified
```

Comment:
given this function is only used in getCgroupSubsystems() i would unroll it there and remove the sync.Once usage and the global variables on line 47.

```
var st unix.Statfs_t
if err := unix.Statfs(unifiedMountpoint, &st); err != nil {
	panic("cannot statfs cgroup root")
}
if st.Type == unix.CGROUP2_SUPER_MAGIC {
	// handle cgroupv2

}
```

Reply:
it might be better to not panic here and instead return this as a:
`return []string, errors.Wrapf(err, "cannot statfs the cgroupv2 root")`

the `err` in this case should already include the path

Reply:
Thank you. I modified this implementation.